File: icopa_cli/icopa_cli/endpoints.py

class Endpoints:
    """ICoPa Hub API endpoints."""

    # Auth
    LOGIN = "auth/user_login/"
    LOGOUT = "auth/user_logout/"

    # Inventory
    INVENTORY = "inventory/"
    INVENTORY_VM = "inventory/vm/"
    INVENTORY_CHECK_VM = "inventory/check_vm/"
    INVENTORY_STOP_CONTAINER = "inventory/stop_container/"

    # Runtime environment
    RUNTIME_ENV = "runtime_env/"
    RUNTIME_ENV_DETAIL = "runtime_env/env/"
    RUNTIME_ENV_ACTIONS = "runtime_env/actions/"

    # Scenario
    SCENARIOS = "scenarios/"
    SCENARIO_DETAIL = "scenarios/scene/"
    SCENARIO_VALIDATE = "scenarios/scene/"

    # Profiling experiments
    PROFILING_EXPS = "profiling_exps/"
    PROFILING_EXP_DETAIL = "profiling_exps/exp/"
    PROFILING_EXP_START = "profiling_exps/exp/"
    PROFILING_EXP_GENERATE = "profiling_exps/exp/"
    PROFILING_EXP_RUN_CHECK = "profiling_exps/exp/"
    PROFILING_EXP_RUNS = "profiling_exps/exp/"
    PROFILING_RUNS = "profiling_exps/runs/"
    PROFILING_RUN_DETAIL = "profiling_exps/runs/"

    # Cloud provisioning
    CLOUD_VMS = "cloud_provisioning/vms/"
    CLOUD_VM_UPLOAD = "cloud_provisioning/vms/upload/"
    CLOUD_VM_CHECK_DETAIL = "cloud_provisioning/vms/{vm_id}/check_requests/{check_id}/"
    CLOUD_VM_START_DETAIL = "cloud_provisioning/vms/{vm_id}/start_requests/{start_id}/"
    CLOUD_VM_STOP_DETAIL = "cloud_provisioning/vms/{vm_id}/stop_requests/{stop_id}/"

File: icopa_gui/src/core/api/endpoints.ts
export const endpoints = {
  auth: {
    login: "auth/user_login/",
    logout: "auth/user_logout/",
  },
  inventory: {
    list: "inventory/",
    detail: (vmName: string) => `inventory/vm/${encodeURIComponent(vmName)}/`,
    check: "inventory/check_vm/",
    stopContainer: "inventory/stop_container/",
  },
  runtime: {
    list: "runtime_env/",
    detail: (envName: string) => `runtime_env/env/${encodeURIComponent(envName)}/`,
  },
  scenario: {
    list: "scenarios/",
    detail: (scenarioName: string) => `scenarios/scene/${encodeURIComponent(scenarioName)}/`,
    validate: (scenarioName: string) => `scenarios/scene/${encodeURIComponent(scenarioName)}/validate/`,
  },
  experiment: {
    list: "profiling_exps/",
    detail: (expName: string) => `profiling_exps/exp/${encodeURIComponent(expName)}/`,
    generate: (expName: string) => `profiling_exps/exp/${encodeURIComponent(expName)}/generate/`,
    start: (expName: string) => `profiling_exps/exp/${encodeURIComponent(expName)}/start/`,
    runsByExperiment: (expName: string) => `profiling_exps/exp/${encodeURIComponent(expName)}/runs/`,
    runList: "profiling_exps/runs/",
    runDetail: (runId: number) => `profiling_exps/runs/${runId}/`,
    runRerun: (runId: number) => `profiling_exps/runs/${runId}/rerun/`,
    comparisionList: "profiling_exps/comparisions/",
  },
  cloud: {
    list: (provider: string) => `cloud_provisioning/vms/?provider=${encodeURIComponent(provider)}`,
    detail: (id: number) => `cloud_provisioning/vms/${id}/`,
    upload: "cloud_provisioning/vms/upload/",
    create: (id: number) => `cloud_provisioning/vms/${id}/create/`,
    delete: (id: number) => `cloud_provisioning/vms/${id}/delete/`,
    start: (id: number) => `cloud_provisioning/vms/${id}/start/`,
    stop: (id: number) => `cloud_provisioning/vms/${id}/stop/`,
    check: (id: number) => `cloud_provisioning/vms/${id}/check/`,
    checkDetail: (id: number, checkId: number) =>
      `cloud_provisioning/vms/${id}/check_requests/${checkId}/`,
    startDetail: (id: number, startId: number) =>
      `cloud_provisioning/vms/${id}/start_requests/${startId}/`,
    stopDetail: (id: number, stopId: number) =>
      `cloud_provisioning/vms/${id}/stop_requests/${stopId}/`,
  },
};
